onnx/backend/test/cpp/driver_test.cc

```diff
@@ -0,0 +1,240 @@
+#include "driver/gtest_utils.h"
+
+#include "onnx/checker.h"
+#include "onnx/onnxifi.h"
+#include "onnx/onnxifi_loader.h"
+
+const bool ONNXIFI_BACKEND_USED = false;
```

Comment:
how is it planned to be used in the future?
Say I have an implementation of the ONNXIFI backend, how would I use the driver to run integration test?

If steps are not obvious it would be good to document.

Reply:
I think you are right. I would need to document this.

Reply:
Is the plan to use #ifdefs here for ONNXIFI_BACKEND_USED?

Reply:
@rdzhabarov In fact we are doing such things:
1. We dont specify onnxifi backend for test driver, but only use by default backend. That does not mean we only use one backend, but means we do not indicate the target backend in test driver, but indicate it in onnxifi_loader. You can check the logic of onnxifi_loader when given empty backend location for more details.
2. As for ONNXIFI_BACKEND_USED, we will finnally enable it when we have valid by default backend to test. Now since we dont have any valid backend, in order to pass gtest, we set this temporary var. We will finnally remove it, or set it to true if by default backend detected.

Reply:
I agree with @rdzhabarov to make ONNXIFI_BACKEND_USED as an ifdef so it can be configured at compile time.


nit: I suggest using ONNXIFI_DUMMY_BACKEND (instead of ONNXIFI_BACKEND_USED) and in your code do
```
#ifndef ONNXIFI_DUMMY_BACKEND
  blablablabla(...)
#endif
```
so that normal real backends don't need to add special -D define to build (at the end it's the dummy backend being the special one to the test driver).